edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -34,7 +34,7 @@ class JwtSessionUserMismatchError(JwtAuthenticationError):
     pass
 
 
-class JwtUserEmailMismatchError(JwtAuthenticationError):
+class JwtUserEmailMismatchError(exceptions.AuthenticationFailed):
```

Comment:
1. Could we change the base class instead?
```
class JwtAuthenticationError(exceptions.AuthenticationFailed):
```
2. Is there any additional assertion we could add to a test to ensure that the right type of exception is being used?

Reply:
> Could we change the base class instead?

Yes, done.

> Is there any additional assertion we could add to a test to ensure that the right type of exception is being used?

Since `AuthenticationFailed` is a known exception to the default exception handler therefore it gets handled by the exception handler and we only get a `401 Unauthorized` response. Exception is not being propagated up and we are not getting it in the test. So, I don't think we can assert the exception type.

Reply:
Although it is unlikely that someone will break this, a unit test that creates an instance of `JwtSessionUserMismatchError` and asserts that it is also an instance of `AuthenticationFailed` would prevent regressions. I'm not sure if it is worth it, but it is also very quick to implement. I'll leave it to you to decide.